Tools/i18n/pygettext.py

```diff
@@ -213,21 +213,39 @@ def escape_nonascii(s, encoding):
     return ''.join(escapes[b] for b in s.encode(encoding))
 
 
-def normalize(s, encoding):
+def normalize(s, encoding, options):
     # This converts the various Python string types into a format that is
-    # appropriate for .po files, namely much closer to C style.
-    lines = s.split('\n')
-    if len(lines) == 1:
-        s = '"' + escape(s, encoding) + '"'
-    else:
-        if not lines[-1]:
-            del lines[-1]
-            lines[-1] = lines[-1] + '\n'
-        for i in range(len(lines)):
-            lines[i] = escape(lines[i], encoding)
-        lineterm = '\\n"\n"'
-        s = '""\n"' + lineterm.join(lines) + '"'
-    return s
+    # appropriate for .po files, namely much closer to C style. While wrapping
+    # to options.width.
+    lines = []
+    for line in s.splitlines(True):
+        if len(escape(line, encoding)) > options.width:
+            words = line.split()
+            words.reverse()
+            while words:
+                buf = []
+                size = 2
+                while words:
+                    word = words[-1]
+                    escaped_word = escape(word, encoding)
+                    add_space = 1 if buf else 0
+                    if size + len(escaped_word) + add_space <= options.width:
+                        buf.append(words.pop())
+                        size += len(escaped_word) + add_space
+                    else:
+                        if not buf:
+                            buf.append(words.pop())
+                        break
+                lines.append(' '.join(buf))
+        else:
+            lines.append(line)
+    if len(lines) <= 1:
+        return '"' + escape(s, encoding) + '"'
+    if lines and not lines[-1]:
+        del lines[-1]
+        lines[-1] += '\n'
+    return '""\n' + '\n'.join(
+        [f'"{escape(line, encoding)}"' for line in lines])
```

Comment:
```suggestion
    escaped_lines = (f'"{escape(line, encoding)}"' for line in lines)
    return '""\n' + '\n'.join(escaped_lines)
```

You could have a small function that does quote + escape btw, say

```py
def _escape_quote(s, encoding):
    return f'"{escape(s, encoding)}"'
```

It could becomes a bit clearer instead of having mixed `'` and `"` everywhere.